$OpenBSD: patch-src_rpc_scgi_cc,v 1.3 2012/07/15 10:20:07 dcoppa Exp $

SCGI m_task queue could overflow since the check was not working
(upstream git commit f5d4924b22f7ebbfef111668405987a1b893a412)

--- src/rpc/scgi.cc.orig	Tue Apr  5 12:26:10 2011
+++ src/rpc/scgi.cc	Thu Oct 20 12:32:42 2011
@@ -139,7 +139,7 @@ SCgi::event_read() {
   while ((fd = get_fd().accept(&sa)).is_valid()) {
     SCgiTask* task = std::find_if(m_task, m_task + max_tasks, std::mem_fun_ref(&SCgiTask::is_available));
 
-    if (task == task + max_tasks) {
+    if (task == m_task + max_tasks) {
       // Ergh... just closing for now.
       fd.close();
       continue;
